### PR TITLE
GIX-1553: Disable button user restricted country

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

SNS projects can restrict swap participation based on the user's country.

# Changes

* Handle the button status "disabled-not-eligible" in the ParticipateButton.
* Implement the `getDeniedCountries` getter helper.

# Tests

* Add test case in ParticipateButton.spec

### CODE REVIEW

**Comment by lmuntaner** [comment]
User from restricted country:

https://github.com/dfinity/nns-dapp/assets/4550653/aae0b331-a5e6-40bd-af77-4ecd12e7162a

User from non-restricted country:

https://github.com/dfinity/nns-dapp/assets/4550653/7bac627f-5d2b-41d8-8e54-05eeb3788199

File: frontend/src/lib/components/project-detail/ParticipateButton.svelte [hunk 1]
```diff
@@ -89,7 +89,17 @@
...
+              text={$i18n.sns_project_detail.not_eligible_to_participate}
+            >
+              <button
+                class="primary"
+                data-tid="sns-project-participate-button"
+                disabled>{$i18n.sns_project_detail.participate}</button
+              >
+            </Tooltip>
           {:else}
             <!-- This is the "enabled" and "disabled-not-eligible" case -->
```
Comment (dskloetd) [inline frontend/src/lib/components/project-detail/ParticipateButton.svelte:104]: This comment needs to be updated because `"disabled-not-eligible"` is now covered above.

File: frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts [hunk 2]
```diff
@@ -260,6 +260,36 @@ describe("ParticipateButton", () => {
...
+      const button = queryByTestId(
+        "sns-project-participate-button"
+      ) as HTMLButtonElement;
+      expect(button.getAttribute("disabled")).not.toBeNull();
+
+      const tooltipPo = new TooltipPo(new JestPageObjectElement(container));
+      expect(await tooltipPo.getText()).toBe(
+        "You are not eligible to participate in this swap."
+      );
+    });
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts:292]: Should we also have a test with a non-empty deny list, but where the user is in a different country and is allowed to participate, to make sure the button then is not disabled?

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/components/project-detail/ParticipateButton.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/components/project-detail/ParticipateButton.svelte:104]: thanks!
